  # **1. Imports**



In [92]:
import numpy as np
import pandas as pd




# **2.   Chargement des datasets**



In [93]:
df_communes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/details_communes.csv")
df_centres = pd.read_csv("Documents/gyj/Test_technique_analyst/data/centres_service.csv")
df_socio = pd.read_csv("Documents/gyj/Test_technique_analyst/data/donnees_socioeconomiques.csv")
df_dev = pd.read_csv("Documents/gyj/Test_technique_analyst/data/developpement.csv")
df_routes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/reseau_routier_togo_ext.csv")
df_docs = pd.read_csv("Documents/gyj/Test_technique_analyst/data/documents_administratifs_ext.csv")
df_demandes = pd.read_csv("Documents/gyj/Test_technique_analyst/data/demandes_service_public.csv")
df_logs = pd.read_csv("Documents/gyj/Test_technique_analyst/data/logs_activite.csv")


# **3. Aperçu structurel**


# **details_communes**

In [94]:
# Vérification des dimensions des dataframes
df_communes.shape

(200, 13)

In [95]:
# Vérification des types de variables
df_communes.dtypes

commune_id               object
commune                  object
prefecture               object
region                   object
latitude                float64
longitude               float64
altitude_m                int64
superficie_km2            int64
population_densite        int64
code_postal              object
type_commune             object
distance_capitale_km      int64
zone_climatique          object
dtype: object

In [96]:
# Aperçu du contenu
df_communes.head()


commune_id     commune prefecture    region  latitude  longitude  \
0     COM001       Vogan         Vo  Maritime    6.2295     1.5040   
1     COM002     Hahotoé         Vo  Maritime    6.6892     1.6588   
2     COM003  Anyronkopé         Vo  Maritime    6.4158     1.2398   
3     COM004        Momé         Vo  Maritime    6.2023     1.6180   
4     COM005     Dagbati         Vo  Maritime    6.3068     1.6507   

   altitude_m  superficie_km2  population_densite code_postal  type_commune  \
0          19             343                  35      BP 272        Rurale   
1          39             398                 210      BP 239  Semi-urbaine   
2          47             215                  76      BP 495        Rurale   
3          41             230                  95       BP 60        Rurale   
4          25             318                  58      BP 235        Rurale   

   distance_capitale_km zone_climatique  
0                    34         Côtière  
1                    79         Côtière  
2                    31         Côtière  
3                    45         Côtière  
4                    52         Côtière

**Aperçu dataset** : COMMUNES\
 **Dimensions :** (200 , 13)\
 **Types :** object + int    
 **Colonnes importantes :** commune_id, commune, prefecture, region, population, type_commune, distance_capitale, latitude, longitude\
 **Notes :** ce dataset répertorie les informations concernant les communes existant au Togo.

# **centres_service.csv**

In [97]:
# Vérification des dimensions des dataframes
df_centres.shape

(55, 16)

In [98]:
# Vérification des types de variables
df_centres.dtypes

centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [99]:
# Aperçu du contenu
df_centres.head()


centre_id                nom_centre type_centre    region prefecture  \
0     CT001  Centre Principal Dapaong   Principal   Savanes       Tône   
1     CT002               Centre Kévé  Secondaire  Maritime        Ave   
2     CT003              Centre Gando  Secondaire   Savanes    Oti-Sud   
3     CT004            Centre Pagouda  Secondaire      Kara      Binah   
4     CT005            Centre Tchamba  Secondaire  Centrale    Tchamba   

   commune           quartier  latitude  longitude  personnel_capacite_jour  \
0  Dapaong      Administratif   10.5920     0.4742                      300   
1     Kévé  Zone industrielle    6.2815     1.2182                       63   
2    Gando  Zone industrielle   10.2629     0.2525                       98   
3  Pagouda      Gare routière    9.4247     1.1698                       78   
4  Tchamba          Carrefour    8.5008     0.9988                       62   

   nombre_guichets heures_ouverture horaire_nuit equipement_numerique  \
0               12      07:30-17:30          Non              Complet   
1                2      08:00-16:00          Non               Limite   
2                2      08:00-16:00          Non              Partiel   
3                6      08:00-16:00          Non              Partiel   
4                3      08:00-16:00          Non               Limite   

  date_ouverture statut_centre  
0     2016-02-14         Actif  
1     2020-09-14         Actif  
2     2015-06-02         Actif  
3     2019-12-03         Actif  
4     2022-10-20         Actif

**Aperçu dataset :** \
**CENTRES Dimensions :** (55 , 16) \
**Types :** object + int + float \
**Colonnes importantes :** centre_id, nom_centre, type_centre, prefecture, region, commune, quartier, latitude,longitude, personnel_capacite_jour, nombre_guichets, equipement_numerique, statut_centre \
**Notes :** ce dataset répertorie les informations concernant les centres de traitement des demandes des documents officiel au Togo.

#**donnees_socioecomiques.csv**

In [100]:
# Vérification des dimensions des dataframes
df_socio.shape

(115, 11)

In [101]:
# Vérification des types de variables
df_socio.dtypes

region                   object
prefecture               object
commune                  object
population                int64
superficie_km2            int64
densite                   int64
taux_urbanisation       float64
taux_alphabétisation    float64
age_median                int64
nombre_menages            int64
revenu_moyen_fcfa         int64
dtype: object

In [102]:
# Aperçu du contenu
df_socio.head()


region prefecture     commune  population  superficie_km2  densite  \
0  Maritime         Vo       Vogan       48777             478      102   
1  Maritime         Vo     Hahotoé       13939             444       31   
2  Maritime         Vo  Anyronkopé       23395             355       65   
3  Maritime         Vo        Momé       19379             302       64   
4  Maritime         Vo     Dagbati       14846             309       48   

   taux_urbanisation  taux_alphabétisation  age_median  nombre_menages  \
0               0.33                  0.76          19            8590   
1               0.35                  0.66          21            3973   
2               0.18                  0.72          21            4534   
3               0.10                  0.81          22            4207   
4               0.32                  0.79          22            3170   

   revenu_moyen_fcfa  
0              78348  
1             112380  
2             107120  
3              99265  
4             133012

**Aperçu dataset :** Socio-economique \
**Dimensions :** (115 , 11) \
**Types :** object + int + float \
**Colonnes importantes :** region, prefecture, commune, population, superficie_km2, densite, taux_urbanisation, taux_alphabétisation, age_median, nombre_menages, revenu_moyen_fcfa \
**Notes** : ce dataset répertorie les informations socioéconomique de la population par région,préfecture et commune au Togo.

# **developpement.csv**

In [103]:
# Vérification des dimensions des dataframes
df_dev.shape

(33, 15)

In [104]:
# Vérification des types de variables
df_dev.dtypes

region                    object
prefecture                object
commune                   object
pib_par_habitant_fcfa      int64
taux_chomage             float64
taux_pauvrete            float64
acces_electricite        float64
acces_eau_potable        float64
acces_internet           float64
indice_developpement     float64
score_education          float64
score_sante              float64
nombre_ecoles              int64
nombre_hopitaux            int64
nombre_banques             int64
dtype: object

In [105]:
# Aperçu du contenu
df_dev.head()


region prefecture   commune  pib_par_habitant_fcfa  taux_chomage  \
0  Maritime         Vo      Lomé                 860000          0.11   
1  Maritime         Vo  Adakpamé                 830000          0.12   
2  Maritime      Golfe        Bè                 720000          0.13   
3  Maritime      Golfe      Agoè                 700000          0.14   
4  Maritime       Lacs     Aného                 560000          0.16   

   taux_pauvrete  acces_electricite  acces_eau_potable  acces_internet  \
0           0.24               0.96               0.93            0.67   
1           0.26               0.94               0.91            0.63   
2           0.27               0.92               0.89            0.60   
3           0.29               0.91               0.88            0.58   
4           0.31               0.86               0.83            0.46   

   indice_developpement  score_education  score_sante  nombre_ecoles  \
0                  0.74             0.79         0.76            250   
1                  0.71             0.77         0.74            220   
2                  0.69             0.75         0.73            190   
3                  0.67             0.73         0.71            175   
4                  0.63             0.69         0.66             98   

   nombre_hopitaux  nombre_banques  
0               19              48  
1               16              40  
2               13              30  
3               12              26  
4                6              16

**Aperçu dataset :** Developpement \
**Dimensions** : (33 , 15) \
**Types :** object + int + float
**Colonnes importantes :** region, prefecture, commune, pib_par_habitant_fcfa, taux_chomage, taux_pauvrete, acces_electricite, acces_eau_potable, acces_internet, indice_developpement, score_education, score_sante, nombre_ecoles, nombre_hopitaux, nombre_banques \
**Notes :** ce dataset contient les indicateurs de mesure du développement par region, prefecture, communel au Togo.

# **reseau_routier_togo_ext.csv**

In [106]:
# Vérification des dimensions des dataframes
df_routes.shape

(40, 14)

In [107]:
# Vérification des types de variables
df_routes.dtypes

route_id                   object
type_route                 object
nom_route                  object
region_origine             object
region_destination         object
prefecture_origine         object
prefecture_destination     object
longueur_km                 int64
etat_route                 object
temps_parcours_heures     float64
points_controle             int64
bus_par_jour                int64
camions_par_jour            int64
passagers_par_jour          int64
dtype: object

In [108]:
# Aperçu du contenu
df_routes.head()


route_id type_route nom_route region_origine region_destination  \
0     R001  Nationale       RN1       Maritime           Plateaux   
1     R002  Nationale       RN2       Plateaux           Centrale   
2     R003  Nationale       RN3       Centrale               Kara   
3     R004  Nationale       RN4           Kara            Savanes   
4     R005  Nationale       RN5       Maritime               Kara   

  prefecture_origine prefecture_destination  longueur_km etat_route  \
0                 Vo                   Ogou          150        Bon   
1               Ogou              Sotouboua          120      Moyen   
2          Sotouboua                  Kozah          110      Moyen   
3              Kozah                    Oti           95   Mediocre   
4                 Vo                  Kozah          420      Moyen   

   temps_parcours_heures  points_controle  bus_par_jour  camions_par_jour  \
0                    2.5                3            85                45   
1                    2.8                2            65                35   
2                    2.6                2            60                30   
3                    3.0                2            45                25   
4                    8.5                6            70                55   

   passagers_par_jour  
0                1250  
1                 950  
2                 850  
3                 650  
4                1600

**Aperçu dataset :** RESEAUX ROUTIERS \
**Dimensions :** (40 , 14) \
**Types :** object + int + float \
**Colonnes importantes :** route_id type_route nom_route region_origine region_destination prefecture_origine prefecture_destination longueur_km etat_route temps_parcours_heures points_controle bus_par_jour camions_par_jour passagers_par_jour \
**Notes :** ce dataset répertorie les informations concernant les réseaux routiers au Togo.

# **demandes_service_public.csv**

In [109]:
# Vérification des dimensions des dataframes
df_demandes.shape

(600, 16)

In [110]:
# Vérification des types de variables
df_demandes.dtypes

demande_id                 object
region                     object
prefecture                 object
commune                    object
quartier                   object
type_document              object
categorie_document         object
nombre_demandes             int64
delai_traitement_jours      int64
taux_rejet                float64
date_demande               object
motif_demande              object
statut_demande             object
canal_demande              object
age_demandeur               int64
sexe_demandeur             object
dtype: object

In [111]:
# Aperçu du contenu
df_demandes.head()


demande_id    region prefecture    commune     quartier  \
0       D001  Centrale  Sotouboua  Sotouboua  Nyékonakpoé   
1       D002      Kara      Binah    Pagouda      Hôpital   
2       D003  Plateaux       Ogou   Atakpamé           Bè   
3       D004   Savanes        Oti      Mango     Amoutivé   
4       D005  Maritime        Zio     Tsévié    Adidogomé   

               type_document categorie_document  nombre_demandes  \
0           Carte d'identité           Identité              168   
1           Carte d'identité           Identité               17   
2  Certificat de nationalité           Identité              104   
3           Carte d'identité           Identité               92   
4           Livre de famille              Civil              138   

   delai_traitement_jours  taux_rejet date_demande     motif_demande  \
0                       1        0.00   2023-09-30             Perte   
1                      22        0.09   2023-12-22               Vol   
2                       4        0.08   2023-08-13          Concours   
3                      39        0.02   2023-06-08  Première demande   
4                       7        0.12   2023-02-02     Détérioration   

  statut_demande canal_demande  age_demandeur sexe_demandeur  
0        Rejetée      En ligne             27              M  
1        Traitee      En ligne             21              F  
2        Traitee      En ligne             18              M  
3        Rejetée       Guichet             64              M  
4        Rejetée      En ligne             30              M

**Aperçu dataset :** DEMANDES SERVICE PUBLIC \
**Dimensions :** (600 , 16) \
**Types :** object + int + float \
**Colonnes importantes :** demande_id region prefecture commune quartier type_document categorie_document nombre_demandes delai_traitement_jours taux_rejet date_demande motif_demande statut_demande canal_demande age_demandeur sexe_demandeur \
**Notes :** ce dataset contient les informations sur chaque demande de documents officiel au Togo.

# **documents_admistratifs.csv**

In [112]:
# Vérification des dimensions des dataframes
df_docs.shape

(64, 9)

In [113]:
# Vérification des types de variables
df_docs.dtypes

annee                  int64
mois                   int64
region                object
prefecture            object
commune               object
type_document         object
nombre_demandes        int64
delai_moyen_jours    float64
taux_rejet_moyen     float64
dtype: object

In [114]:
# Aperçu du contenu
df_docs.head()


annee  mois    region prefecture commune      type_document  \
0   2023     1  Maritime         Vo    Lomé   Carte d'identité   
1   2023     1  Maritime         Vo    Lomé          Passeport   
2   2023     1  Maritime         Vo    Lomé  Acte de naissance   
3   2023     1  Maritime         Vo    Lomé   Livre de famille   
4   2023     1  Maritime       Lacs   Aného   Carte d'identité   

   nombre_demandes  delai_moyen_jours  taux_rejet_moyen  
0             2450               12.5              0.05  
1             1650               25.2              0.08  
2              890               10.1              0.02  
3              620               15.3              0.03  
4             1250               14.5              0.04

**Aperçu dataset :** DOCUMENTS ADMINISTRATIFS \
**Dimensions :** (64, 9) \
**Types :** object + int + float \
**Colonnes importantes :** annee mois region prefecture commune type_document nombre_demandes delai_moyen_jours taux_rejet_moyen \
**Notes :** ce dataset contient les informations agrégées sur les demandes par année, mois..., des documents officiel au Togo.

# **logs_activites.csv**

In [115]:
# Vérification des dimensions des dataframes
df_logs.shape

(450, 14)

In [116]:
# Vérification des types de variables
df_logs.dtypes

log_id                         object
centre_id                      object
date_operation                 object
type_operation                 object
type_document                  object
nombre_traite                   int64
delai_effectif                  int64
nombre_rejete                   int64
raison_rejet                   object
personnel_present               int64
temps_attente_moyen_minutes     int64
incident_technique             object
heure_debut                    object
heure_fin                      object
dtype: object

In [117]:
# Aperçu du contenu
df_logs.head()


log_id centre_id date_operation type_operation      type_document  \
0  LOG001     CT015     2023-10-30     Traitement   Carte d'identité   
1  LOG002     CT026     2023-04-21    Maintenance                NaN   
2  LOG003     CT049     2023-08-22     Traitement   Livre de famille   
3  LOG004     CT047     2023-07-13     Traitement  Acte de naissance   
4  LOG005     CT008     2023-04-08     Traitement          Passeport   

   nombre_traite  delai_effectif  nombre_rejete    raison_rejet  \
0            241              41             31    Non éligible   
1              0               0              0             NaN   
2             62              19              0             NaN   
3            139              22             12  Double demande   
4             71              35              6  Double demande   

   personnel_present  temps_attente_moyen_minutes incident_technique  \
0                 15                           58                Non   
1                 12                            0                Non   
2                 12                           95                Non   
3                  7                           54                Non   
4                 13                           94                Oui   

  heure_debut heure_fin  
0       08:00     16:00  
1       09:00     19:00  
2       08:30     17:30  
3       08:15     16:15  
4       08:30     17:30

**Aperçu dataset :** LOGS ACTIVITES \
**Dimensions :** (450 , 14) \
**Types :** object + int + float \
**Colonnes importantes :** log_id centre_id date_operation type_operation type_document nombre_traite delai_effectif nombre_rejete raison_rejet personnel_present temps_attente_moyen_minutes incident_technique heure_debut heure_fin \
**Notes** : ce dataset contient les informations de logs des traitements des demandes des documents officiel au Togo.

# **4. Identification des clés**


# **5. Notes et observations**


# **BLOC 2: Qualité des données**


# **details_communes**

In [120]:
print("Dimensions :", df_communes.shape)
print("Types des colonnes :")
df_communes.dtypes


Dimensions : (200, 13)
Types :


commune_id               object
commune                  object
prefecture               object
region                   object
latitude                float64
longitude               float64
altitude_m                int64
superficie_km2            int64
population_densite        int64
code_postal              object
type_commune             object
distance_capitale_km      int64
zone_climatique          object
dtype: object

In [123]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_communes.isnull().sum()


Valeurs manquantes :


commune_id              0
commune                 0
prefecture              0
region                  0
latitude                0
longitude               0
altitude_m              0
superficie_km2          0
population_densite      0
code_postal             0
type_commune            0
distance_capitale_km    0
zone_climatique         0
dtype: int64

In [125]:
#Vérifier les doublons
print("Doublons :", df_communes.duplicated().sum())


Doublons : 0


In [127]:
# Vérifier la granularité des données
df_communes.nunique()

commune_id              200
commune                 198
prefecture               37
region                    5
latitude                199
longitude               197
altitude_m              164
superficie_km2          161
population_densite      126
code_postal             160
type_commune              3
distance_capitale_km    164
zone_climatique           4
dtype: int64

In [128]:
# Décrire les valeurs numériques
df_communes.describe()


latitude   longitude  altitude_m  superficie_km2  population_densite  \
count  200.000000  200.000000  200.000000      200.000000          200.000000   
mean     8.153211    1.059142  290.525000      362.885000          208.610000   
std      1.513130    0.365370  205.517827      130.180477          569.698243   
min      6.112500    0.006700    8.000000      102.000000           20.000000   
25%      6.652650    0.840575   67.000000      249.750000           43.750000   
50%      7.759900    1.150050  283.500000      351.500000           77.000000   
75%      9.561700    1.314600  443.500000      479.000000          155.250000   
max     11.032300    1.688700  783.000000      600.000000         4690.000000   

       distance_capitale_km  
count            200.000000  
mean             229.715000  
std              166.393624  
min                3.000000  
25%               67.250000  
50%              183.500000  
75%              381.000000  
max              556.000000

In [129]:
# Décrire les variables catégorielles
df_communes.describe(include=['object'])

commune_id           commune prefecture    region code_postal  \
count         200               200        200       200         200   
unique        200               198         37         5         160   
top        COM001  Cinkassé Central      Golfe  Maritime      BP 239   
freq            1                 2         11        55           3   

       type_commune zone_climatique  
count           200             200  
unique            3               4  
top          Rurale          Savane  
freq            138              87

In [131]:
cat_cols = df_communes.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_communes[col].unique())

commune_id => ['COM001' 'COM002' 'COM003' 'COM004' 'COM005' 'COM006' 'COM007' 'COM008'
 'COM009' 'COM010' 'COM011' 'COM012' 'COM013' 'COM014' 'COM015' 'COM016'
 'COM017' 'COM018' 'COM019' 'COM020' 'COM021' 'COM022' 'COM023' 'COM024'
 'COM025' 'COM026' 'COM027' 'COM028' 'COM029' 'COM030' 'COM031' 'COM032'
 'COM033' 'COM034' 'COM035' 'COM036' 'COM037' 'COM038' 'COM039' 'COM040'
 'COM041' 'COM042' 'COM043' 'COM044' 'COM045' 'COM046' 'COM047' 'COM048'
 'COM049' 'COM050' 'COM051' 'COM052' 'COM053' 'COM054' 'COM055' 'COM056'
 'COM057' 'COM058' 'COM059' 'COM060' 'COM061' 'COM062' 'COM063' 'COM064'
 'COM065' 'COM066' 'COM067' 'COM068' 'COM069' 'COM070' 'COM071' 'COM072'
 'COM073' 'COM074' 'COM075' 'COM076' 'COM077' 'COM078' 'COM079' 'COM080'
 'COM081' 'COM082' 'COM083' 'COM084' 'COM085' 'COM086' 'COM087' 'COM088'
 'COM089' 'COM090' 'COM091' 'COM092' 'COM093' 'COM094' 'COM095' 'COM096'
 'COM097' 'COM098' 'COM099' 'COM100' 'COM101' 'COM102' 'COM103' 'COM104'
 'COM105' 'COM106' 'COM107' 'COM108' 

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : non

* Observations : Vérification unicité (commune_id=200, commune=198), COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : non

* Observations : Vérification unicité (commune_id=200, commune=198); la commune "Cinkassé Central" apparaît 2 fois

# **centres_service.csv**

In [132]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [133]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [134]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [135]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [137]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [139]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [141]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

#**donnees_socioecomiques.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

# **developpement.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

# **reseau_routier_togo_ext.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

# **demandes_service_public.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

# **documents_admistratifs.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)

# **logs_activites.csv**

In [ ]:
print("Dimensions :", df_centres.shape)
print("Types des colonnes :")
df_centres.dtypes


Dimensions : (55, 16)
Types des colonnes :


centre_id                   object
nom_centre                  object
type_centre                 object
region                      object
prefecture                  object
commune                     object
quartier                    object
latitude                   float64
longitude                  float64
personnel_capacite_jour      int64
nombre_guichets              int64
heures_ouverture            object
horaire_nuit                object
equipement_numerique        object
date_ouverture              object
statut_centre               object
dtype: object

In [ ]:
# Vérifier les valeurs manquantes
print("Valeurs manquantes :")
df_centres.isnull().sum()


Valeurs manquantes :


centre_id                  0
nom_centre                 0
type_centre                0
region                     0
prefecture                 0
commune                    0
quartier                   0
latitude                   0
longitude                  0
personnel_capacite_jour    0
nombre_guichets            0
heures_ouverture           0
horaire_nuit               0
equipement_numerique       0
date_ouverture             0
statut_centre              0
dtype: int64

In [ ]:
#Vérifier les doublons
print("Doublons :", df_centres.duplicated().sum())


Doublons : 0


In [ ]:
# Vérifier la granularité des données
df_centres.nunique()

centre_id                  55
nom_centre                 31
type_centre                 2
region                      5
prefecture                 28
commune                    28
quartier                   10
latitude                   55
longitude                  55
personnel_capacite_jour    44
nombre_guichets            11
heures_ouverture            2
horaire_nuit                2
equipement_numerique        3
date_ouverture             53
statut_centre               1
dtype: int64

In [ ]:
# Décrire les valeurs numériques
df_centres.describe()


latitude  longitude  personnel_capacite_jour  nombre_guichets
count  55.000000  55.000000                55.000000        55.000000
mean    8.345384   1.061744               126.418182         5.327273
std     1.492890   0.336674                95.720599         3.564509
min     6.214500   0.098700                32.000000         2.000000
25%     6.587650   0.844950                63.000000         2.000000
50%     8.685100   1.131300                88.000000         5.000000
75%     9.422350   1.300600               116.500000         6.000000
max    11.002200   1.680600               358.000000        14.000000

In [ ]:
# Décrire les variables catégorielles
df_centres.describe(include=['object'])

centre_id      nom_centre type_centre    region prefecture  commune  \
count         55              55          55        55         55       55   
unique        55              31           2         5         28       28   
top        CT001  Centre Tchamba  Secondaire  Centrale    Tchamba  Tchamba   
freq           1               5          43        16          5        5   

           quartier heures_ouverture horaire_nuit equipement_numerique  \
count            55               55           55                   55   
unique           10                2            2                    3   
top     Résidentiel      08:00-16:00          Non              Partiel   
freq              8               43           52                   31   

       date_ouverture statut_centre  
count              55            55  
unique             53             1  
top        2020-08-10         Actif  
freq                2            55

In [ ]:
cat_cols = df_centres.select_dtypes(include=['object']).columns
for col in cat_cols:
  print (col, "=>", df_centres[col].unique())

centre_id => ['CT001' 'CT002' 'CT003' 'CT004' 'CT005' 'CT006' 'CT007' 'CT008' 'CT009'
 'CT010' 'CT011' 'CT012' 'CT013' 'CT014' 'CT015' 'CT016' 'CT017' 'CT018'
 'CT019' 'CT020' 'CT021' 'CT022' 'CT023' 'CT024' 'CT025' 'CT026' 'CT027'
 'CT028' 'CT029' 'CT030' 'CT031' 'CT032' 'CT033' 'CT034' 'CT035' 'CT036'
 'CT037' 'CT038' 'CT039' 'CT040' 'CT041' 'CT042' 'CT043' 'CT044' 'CT045'
 'CT046' 'CT047' 'CT048' 'CT049' 'CT050' 'CT051' 'CT052' 'CT053' 'CT054'
 'CT055']
nom_centre => ['Centre Principal Dapaong' 'Centre Kévé' 'Centre Gando' 'Centre Pagouda'
 'Centre Tchamba' 'Centre Principal Tsévié' 'Centre Bassar'
 'Centre Principal Kpalimé' 'Centre Principal Kara' 'Centre Sotouboua'
 'Centre Tohoun' 'Centre Danyi-Apéyémé' 'Centre Lomé' 'Centre Tandjouaré'
 'Centre Bafilo' 'Centre Aného' 'Centre Vogan' 'Centre Djarkpanga'
 'Centre Principal Sokodé' 'Centre Naki-Est' 'Centre Blitta-Gare'
 'Centre Dapaong' 'Centre Tabligbo' 'Centre Principal Lomé'
 'Centre Guérin-Kouka' 'Centre Sokodé' 'Centre Niamto

COMMUNES :

* Manquants : 0

* Doublons : 0

* Types incohérents : date_ouverture=object

* Observations : Vérification unicité (centre_id=55, nom_centre=31)